Jenkinsfile

```diff
@@ -1,216 +1,123 @@
 // vim: ft=groovy
 
-// Helpers for setting commit status
-def getRepoUrl() {
-    return sh(script: "git config --get remote.origin.url", returnStdout: true).trim()
-}
-
-def getCommitSha() {
-    return sh(script: "git rev-parse HEAD", returnStdout: true).trim()
-}
-
-
-def setGitHubBuildStatus(status) {
-    def repoUrl = getRepoUrl()
-    def commitSha = getCommitSha()
-
-    step([
-        $class: 'GitHubCommitStatusSetter',
-        reposSource: [$class: "ManuallyEnteredRepositorySource", url: repoUrl],
-        commitShaSource: [$class: "ManuallyEnteredShaSource", sha: commitSha],
-        errorHandlers: [[$class: 'ShallowAnyErrorHandler']],
-        statusResultSource: [
-          $class: 'ConditionalStatusResultSource',
-          results: [
-            [$class: 'BetterThanOrEqualBuildResult', result: 'SUCCESS', state: 'SUCCESS', message: status],
-            [$class: 'BetterThanOrEqualBuildResult', result: 'FAILURE', state: 'FAILURE', message: status],
-            [$class: 'AnyBuildResult', state: 'FAILURE', message: 'Loophole']
-          ]
-        ]
-    ])
-}
+def NUM_CORES = 8
+def errors = []
+def labels = ['CentOS', 'Fedora', 'Ubuntu', 'Mac'] // labels for Jenkins node types we will build on
+def nodes = [:] 
+
+for (lbl in labels) {
+    def label = lbl
+    def envFile = (label == "CentOS") ? "environment_gcc4.yml" : "environment.yml"
+
+    nodes[label] = {
+        stage(label) {
+            isisNode(label) {
+                def isisEnv = [
+                    "ISIS3DATA=/isisData/data",
+                    "ISIS3TESTDATA=/isisData/testData",
+                    "ISIS3MGRSCRIPTS=/isisData/data/isis3mgr_scripts",
+                    "MALLOC_CHECK_=1"
+                ]
+
+                def cmakeFlags = [
+                    "-DJP2KFLAG=ON",
+                    "-DKAKADU_INCLUDE_DIR=/isisData/kakadu",
+                    "-Dpybindings=OFF",
+                    "-DCMAKE_BUILD_TYPE=RELEASE"
+                ]
+
+                def stageStatus = "Checking out ISIS on ${label}"
+
+                // Checkout
+                checkout scm
+
+                if (label == "Mac") {
```

Comment:
> Mac failures seem like mostly a developer problem:
> 
> ```
> /Users/jenkins/workspace/USGS-Astrogeology/ISIS3/PR-3720/build/make/config.darwin:12: /Users/jenkins/workspace/USGS-Astrogeology/ISIS3/PR-3720/build/make/config.darwin-MacOSX10_15: No such file or directory
> 337: /Users/jenkins/workspace/USGS-Astrogeology/ISIS3/PR-3720/build/make/isismake.os:62: *** Unsupported platform, can not make for Darwin.  Stop.
> ```
> 
> Basically, this thing is running a newer version of mac, you can probably do a `cp ISIS3/isis/make/config.darwin-MacOSX10_13 ISIS3/isis/make/config.darwin-MacOSX10_15` and solve the problem. If that simple change doesn't fix it then it's probably something we need to handle on our own after this PR.
> 
> The Embree failure doesn't make much sense unless the underlying OS's for Linux boxes changed. Not sure what spice server is about either...

Added here

Reply:
you want to do that in your repo and commit the new file and push as it's part of the source tree.

Reply:
Oh ok will do